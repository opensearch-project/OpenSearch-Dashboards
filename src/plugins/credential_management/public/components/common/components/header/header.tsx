/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';

import { EuiBetaBadge, EuiSpacer, EuiTitle, EuiText, EuiCode, EuiLink } from '@elastic/eui';

import { i18n } from '@osd/i18n';
import { FormattedMessage } from '@osd/i18n/react';
import { DocLinksStart } from 'src/core/public';
import { useOpenSearchDashboards } from '../../../../../../opensearch_dashboards_react/public';
import { CredentialManagementContext } from '../../../../types';

export const Header = ({
  prompt,
  docLinks,
  headerTitle,
  isBeta = false,
  isCreateCredential = false,
}: {
  prompt?: React.ReactNode;
  docLinks?: DocLinksStart;
  headerTitle: string;
  isBeta?: boolean;
  isCreateCredential: boolean;
}) => {
  const changeTitle = useOpenSearchDashboards<CredentialManagementContext>().services.chrome
    .docTitle.change;

  changeTitle(headerTitle);

  return (
    <div>
      <EuiTitle>
        <h1>
          {headerTitle}

          {isBeta ? (
            <>
              <EuiBetaBadge
                label={i18n.translate('credentialManagement.createCredential.betaLabel', {
                  defaultMessage: 'Beta',
                })}
              />
            </>
          ) : null}
        </h1>
      </EuiTitle>
      <EuiSpacer size="s" />

      {isCreateCredential ? (
        <>
          <EuiText>
            <p>
              <FormattedMessage
                id="credentialManagement.createCredential.description"
                defaultMessage="A credential can be attached to multiple sources. For example, {credential} can be attached to two data sources {first} and {second}."
                values={{
                  credential: <EuiCode>username-password-credential</EuiCode>,
                  first: <EuiCode>os-service-log</EuiCode>,
                  second: <EuiCode>os-application-log</EuiCode>,
                }}
              />
              <br />

              <EuiLink
                href={docLinks?.links.noDocumentation.indexPatterns.introduction}
                target="_blank"
                external
              >
                <FormattedMessage
                  id="credentialManagement.createCredential.documentation"
                  defaultMessage="Read documentation"
                />
              </EuiLink>
            </p>
          </EuiText>
        </>
      ) : null}

      {prompt ? (
        <>
          <EuiSpacer size="m" />
          {prompt}
        </>
      ) : null}
    </div>
  );
};
