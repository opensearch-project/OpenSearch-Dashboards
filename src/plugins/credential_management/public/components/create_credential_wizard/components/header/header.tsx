/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';

import { EuiBetaBadge, EuiSpacer, EuiTitle, EuiText, EuiCode, EuiLink } from '@elastic/eui';

import { i18n } from '@osd/i18n';
import { FormattedMessage } from '@osd/i18n/react';
import { DocLinksStart } from 'opensearch-dashboards/public';
import { useOpenSearchDashboards } from '../../../../../../opensearch_dashboards_react/public';
import { CredentialManagementContext } from '../../../../types';
export const Header = ({ docLinks }: { docLinks: DocLinksStart }) => {
  const changeTitle = useOpenSearchDashboards<CredentialManagementContext>().services.chrome
    .docTitle.change;
  const createCredentialHeader = i18n.translate('credentialManagement.createIndexPatternHeader', {
    defaultMessage: 'Save Your Credential',
  });

  changeTitle(createCredentialHeader);

  return (
    <div>
      <EuiTitle>
        <h1>
          {createCredentialHeader}
          <>
            {' '}
            <EuiBetaBadge
              label={i18n.translate('credentialManagement.createCredential.betaLabel', {
                defaultMessage: 'Beta',
              })}
            />
          </>
        </h1>
      </EuiTitle>
      <EuiSpacer size="s" />
      <EuiText>
        <p>
          <FormattedMessage
            id="credentialManagement.createCredential.description"
            defaultMessage="A credential can be attached to multiple sources. For example, {credential} can be attached to two data sources {first} and {second}."
            values={{
              credential: <EuiCode>username-password-credential</EuiCode>,
              first: <EuiCode>os-service-log</EuiCode>,
              second: <EuiCode>os-application-log</EuiCode>,
            }}
          />
          <br />
          <EuiLink
            href={docLinks.links.noDocumentation.indexPatterns.introduction}
            target="_blank"
            external
          >
            <FormattedMessage
              id="credentialManagement.createCredential.documentation"
              defaultMessage="Read documentation"
            />
          </EuiLink>
        </p>
      </EuiText>
      {prompt ? (
        <>
          <EuiSpacer size="m" />
          {prompt}
        </>
      ) : null}
    </div>
  );
};
