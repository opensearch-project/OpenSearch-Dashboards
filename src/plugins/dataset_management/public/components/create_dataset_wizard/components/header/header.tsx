/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';

import { EuiBetaBadge, EuiSpacer, EuiText } from '@elastic/eui';

import { i18n } from '@osd/i18n';
import { DocLinksStart } from 'opensearch-dashboards/public';
import { useOpenSearchDashboards } from '../../../../../../opensearch_dashboards_react/public';
import { DatasetManagmentContext } from '../../../../types';
import { Description } from './description';

export const Header = ({
  prompt,
  datasetName,
  isBeta = false,
  docLinks,
}: {
  prompt?: React.ReactNode;
  datasetName: string;
  isBeta?: boolean;
  docLinks: DocLinksStart;
}) => {
  const changeTitle = useOpenSearchDashboards<DatasetManagmentContext>().services.chrome.docTitle
    .change;
  const createDatasetHeader = i18n.translate('datasetManagement.createDatasetHeader', {
    defaultMessage: 'Create {datasetName}',
    values: { datasetName },
  });

  changeTitle(createDatasetHeader);

  return (
    <div>
      <EuiText size="s">
        <h1>
          {createDatasetHeader}
          {isBeta ? (
            <>
              {' '}
              <EuiBetaBadge
                label={i18n.translate('datasetManagement.createDataset.betaLabel', {
                  defaultMessage: 'Beta',
                })}
              />
            </>
          ) : null}
        </h1>
      </EuiText>
      <EuiSpacer size="s" />
      <Description docLinks={docLinks} />
      {prompt ? (
        <>
          <EuiSpacer size="m" />
          {prompt}
        </>
      ) : null}
    </div>
  );
};
