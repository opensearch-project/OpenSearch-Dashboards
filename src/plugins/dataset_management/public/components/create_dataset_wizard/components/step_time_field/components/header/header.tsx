/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';

import { EuiTitle, EuiSpacer, EuiText } from '@elastic/eui';

import { FormattedMessage } from '@osd/i18n/react';
import { StepInfo } from '../../../../types';

interface HeaderProps {
  dataset: string;
  datasetName: string;
  stepInfo: StepInfo;
}

export const Header: React.FC<HeaderProps> = ({ dataset, datasetName, stepInfo }) => (
  <div>
    <EuiTitle size="s">
      <h2>
        <FormattedMessage
          id="datasetManagement.createDataset.stepTimeHeader"
          defaultMessage="Step {currentStepNumber} of {totalStepNumber}: Configure settings"
          values={{
            currentStepNumber: stepInfo.currentStepNumber,
            totalStepNumber: stepInfo.totalStepNumber,
          }}
        />
      </h2>
    </EuiTitle>
    <EuiSpacer size="m" />
    <EuiText>
      <FormattedMessage
        id="datasetManagement.createDataset.stepTimeLabel"
        defaultMessage="Specify settings for your {dataset} {datasetName}."
        values={{
          dataset: <strong>{dataset}</strong>,
          datasetName,
        }}
      />
    </EuiText>
  </div>
);
