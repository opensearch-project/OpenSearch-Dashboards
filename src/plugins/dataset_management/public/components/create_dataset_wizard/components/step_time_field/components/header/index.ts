/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { Header } from './header';
