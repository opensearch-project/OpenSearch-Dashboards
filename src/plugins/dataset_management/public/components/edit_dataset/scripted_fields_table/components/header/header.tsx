/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { EuiSmallButton, EuiFlexGroup, EuiFlexItem, EuiText, EuiTitle } from '@elastic/eui';

import { FormattedMessage } from '@osd/i18n/react';
import { ScopedHistory } from 'opensearch-dashboards/public';

import { reactRouterNavigate } from '../../../../../../../opensearch_dashboards_react/public';

interface HeaderProps extends RouteComponentProps {
  datasetId: string;
  history: ScopedHistory;
  useUpdatedUX: boolean;
}

export const Header = withRouter(({ datasetId, history, useUpdatedUX }: HeaderProps) => (
  <EuiFlexGroup alignItems="center">
    <EuiFlexItem>
      <EuiTitle size="s">
        <h3>
          <FormattedMessage
            id="datasetManagement.editDataset.scriptedHeader"
            defaultMessage="Scripted fields"
          />
        </h3>
      </EuiTitle>
      <EuiText size="s">
        <p>
          <FormattedMessage
            id="datasetManagement.editDataset.scriptedLabel"
            defaultMessage="You can use scripted fields in visualizations and display them in your documents. However, you cannot search
            scripted fields."
          />
        </p>
      </EuiText>
    </EuiFlexItem>

    <EuiFlexItem grow={false}>
      <EuiSmallButton
        {...(useUpdatedUX ? { iconType: 'plusInCircle' } : {})}
        data-test-subj="addScriptedFieldLink"
        {...reactRouterNavigate(history, `patterns/${datasetId}/create-field/`)}
      >
        <FormattedMessage
          id="datasetManagement.editDataset.scripted.addFieldButton"
          defaultMessage="Add scripted field"
        />
      </EuiSmallButton>
    </EuiFlexItem>
  </EuiFlexGroup>
));
