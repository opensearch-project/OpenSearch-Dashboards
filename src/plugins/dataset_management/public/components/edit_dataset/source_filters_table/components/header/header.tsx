/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from 'react';

import { EuiTitle, EuiText, EuiSpacer } from '@elastic/eui';
import { FormattedMessage } from '@osd/i18n/react';

export const Header = () => (
  <>
    <EuiTitle size="s">
      <h3>
        <FormattedMessage
          id="datasetManagement.editDataset.sourceHeader"
          defaultMessage="Source filters"
        />
      </h3>
    </EuiTitle>
    <EuiText size="s">
      <p>
        <FormattedMessage
          id="datasetManagement.editDataset.sourceLabel"
          defaultMessage="Source filters can be used to exclude one or more fields when fetching the document source. This happens when
          viewing a document in the Discover app, or with a table displaying results from a saved search in the Dashboard app. Each row is
          built using the source of a single document, and if you have documents with large or unimportant fields you may benefit from
          filtering those out at this lower level."
        />
      </p>
      <p>
        <FormattedMessage
          id="datasetManagement.editDataset.source.noteLabel"
          defaultMessage="Note that multi-fields will incorrectly appear as matches in the table below. These filters only actually apply
          to fields in the original source document, so matching multi-fields are not actually being filtered."
        />
      </p>
    </EuiText>
    <EuiSpacer size="s" />
  </>
);
