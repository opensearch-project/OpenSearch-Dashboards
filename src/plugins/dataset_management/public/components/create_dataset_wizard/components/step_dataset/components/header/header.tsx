/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import './header.scss';

import {
  EuiTitle,
  EuiFlexGroup,
  EuiFlexItem,
  EuiSpacer,
  EuiSmallButton,
  EuiForm,
  EuiCompressedFormRow,
  EuiCompressedFieldText,
  EuiSwitchEvent,
  EuiCompressedSwitch,
} from '@elastic/eui';

import { i18n } from '@osd/i18n';
import { FormattedMessage } from '@osd/i18n/react';
import {
  DataSourceRef,
  DatasetManagmentContext,
} from 'src/plugins/dataset_management/public/types';
import { StepInfo } from '../../../../types';
import { useOpenSearchDashboards } from '../../../../../../../../../plugins/opensearch_dashboards_react/public';

interface HeaderProps {
  isInputInvalid: boolean;
  errors: any;
  characterList: string;
  query: string;
  onQueryChanged: (e: React.ChangeEvent<HTMLInputElement>) => void;
  goToNextStep: (query: string) => void;
  isNextStepDisabled: boolean;
  showSystemIndices?: boolean;
  onChangeIncludingSystemIndices: (event: EuiSwitchEvent) => void;
  isIncludingSystemIndices: boolean;
  stepInfo: StepInfo;
  dataSourceRef?: DataSourceRef;
}

export const Header: React.FC<HeaderProps> = ({
  isInputInvalid,
  errors,
  characterList,
  query,
  onQueryChanged,
  goToNextStep,
  isNextStepDisabled,
  showSystemIndices = false,
  onChangeIncludingSystemIndices,
  isIncludingSystemIndices,
  stepInfo,
  dataSourceRef,
  ...rest
}) => {
  const { dataSourceEnabled } = useOpenSearchDashboards<DatasetManagmentContext>().services;

  return (
    <div {...rest}>
      <EuiTitle size="s">
        <h2>
          <FormattedMessage
            id="datasetManagement.createDataset.stepHeader"
            defaultMessage="Step {currentStepNumber} of {totalStepNumber}: Define an index pattern"
            values={{
              currentStepNumber: stepInfo.currentStepNumber,
              totalStepNumber: stepInfo.totalStepNumber,
            }}
          />
        </h2>
      </EuiTitle>
      <EuiSpacer size="m" />
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiForm isInvalid={isInputInvalid}>
            {dataSourceEnabled && dataSourceRef?.title
              ? renderDataSourceAndDatasetInput(
                  isInputInvalid,
                  errors,
                  characterList,
                  query,
                  onQueryChanged,
                  dataSourceRef.title
                )
              : renderDatasetInput(isInputInvalid, errors, characterList, query, onQueryChanged)}
            {showSystemIndices ? (
              <EuiCompressedFormRow>
                <EuiCompressedSwitch
                  label={
                    <FormattedMessage
                      id="datasetManagement.createDataset.includeSystemIndicesToggleSwitchLabel"
                      defaultMessage="Include system and hidden indices"
                    />
                  }
                  id="checkboxShowSystemIndices"
                  checked={isIncludingSystemIndices}
                  onChange={onChangeIncludingSystemIndices}
                  data-test-subj="showSystemAndHiddenIndices"
                />
              </EuiCompressedFormRow>
            ) : null}
          </EuiForm>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiCompressedFormRow hasEmptyLabelSpace>
            <EuiSmallButton
              fill
              iconSide="right"
              iconType="arrowRight"
              onClick={() => goToNextStep(query)}
              isDisabled={isNextStepDisabled}
              data-test-subj="createDatasetGoToStep2Button"
            >
              <FormattedMessage
                id="datasetManagement.createDataset.step.nextStepButton"
                defaultMessage="Next step"
              />
            </EuiSmallButton>
          </EuiCompressedFormRow>
        </EuiFlexItem>
      </EuiFlexGroup>
    </div>
  );
};

const renderDatasetInput = (
  isInputInvalid: boolean,
  errors: any,
  characterList: string,
  query: string,
  onQueryChanged: (e: React.ChangeEvent<HTMLInputElement>) => void
) => {
  return (
    <EuiCompressedFormRow
      fullWidth
      label={
        <FormattedMessage
          id="datasetManagement.createDataset.step.datasetLabel"
          defaultMessage="Index pattern name"
        />
      }
      isInvalid={isInputInvalid}
      error={errors}
      helpText={
        <>
          <FormattedMessage
            id="datasetManagement.createDataset.step.dataset.allowLabel"
            defaultMessage="Use an asterisk ({asterisk}) to match multiple indices."
            values={{ asterisk: <strong>*</strong> }}
          />{' '}
          <FormattedMessage
            id="datasetManagement.createDataset.step.dataset.disallowLabel"
            defaultMessage="Spaces and the characters {characterList} are not allowed."
            values={{ characterList: <strong>{characterList}</strong> }}
          />
        </>
      }
    >
      <EuiCompressedFieldText
        name="dataset"
        placeholder={i18n.translate('datasetManagement.createDataset.step.datasetPlaceholder', {
          defaultMessage: 'index-name-*',
        })}
        value={query}
        isInvalid={isInputInvalid}
        onChange={onQueryChanged}
        data-test-subj="createDatasetNameInput"
        fullWidth
      />
    </EuiCompressedFormRow>
  );
};

const renderDataSourceAndDatasetInput = (
  isInputInvalid: boolean,
  errors: any,
  characterList: string,
  query: string,
  onQueryChanged: (e: React.ChangeEvent<HTMLInputElement>) => void,
  dataSourceTitle: string
) => {
  return (
    <EuiFlexGroup gutterSize="none">
      <EuiFlexItem grow={2}>
        <EuiCompressedFormRow
          label={
            <FormattedMessage
              id="datasetManagement.createDataset.step.dataSourceLabel"
              defaultMessage="Data Source"
            />
          }
          isInvalid={isInputInvalid}
          error={errors}
        >
          <EuiCompressedFieldText
            name="dataSource"
            placeholder={i18n.translate(
              'datasetManagement.createDataset.step.dataSourcePlaceholder',
              {
                defaultMessage: 'Data source name',
              }
            )}
            value={dataSourceTitle}
            isInvalid={isInputInvalid}
            disabled={true}
            data-test-subj="createDatasetDataSourceName"
          />
        </EuiCompressedFormRow>
      </EuiFlexItem>
      <div className="dataSourceDatasetDot"> {`.`} </div>
      <EuiFlexItem grow={7}>
        {renderDatasetInput(isInputInvalid, errors, characterList, query, onQueryChanged)}
      </EuiFlexItem>
    </EuiFlexGroup>
  );
};
