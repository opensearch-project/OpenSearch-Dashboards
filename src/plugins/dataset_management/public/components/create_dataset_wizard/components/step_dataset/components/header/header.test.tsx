/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { Header } from '../header';
import { shallowWithI18nProvider } from 'test_utils/enzyme_helpers';

describe('Header', () => {
  it('should render normally', () => {
    const component = shallowWithI18nProvider(
      <Header
        isInputInvalid={false}
        errors={[]}
        characterList={'%'}
        query={'k'}
        onQueryChanged={() => {}}
        goToNextStep={() => {}}
        isNextStepDisabled={false}
        onChangeIncludingSystemIndices={() => {}}
        isIncludingSystemIndices={false}
        stepInfo={{ totalStepNumber: 0, currentStepNumber: 0 }}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('should mark the input as invalid', () => {
    const component = shallowWithI18nProvider(
      <Header
        isInputInvalid={true}
        errors={['Input is invalid']}
        characterList={'%'}
        query={'%'}
        onQueryChanged={() => {}}
        goToNextStep={() => {}}
        isNextStepDisabled={true}
        onChangeIncludingSystemIndices={() => {}}
        isIncludingSystemIndices={false}
        stepInfo={{ totalStepNumber: 0, currentStepNumber: 0 }}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
