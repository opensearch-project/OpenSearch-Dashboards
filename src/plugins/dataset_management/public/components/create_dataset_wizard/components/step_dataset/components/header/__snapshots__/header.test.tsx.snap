// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should mark the input as invalid 1`] = `
<div>
  <EuiTitle
    size="s"
  >
    <h2>
      <FormattedMessage
        defaultMessage="Step {currentStepNumber} of {totalStepNumber}: Define an index pattern"
        id="datasetManagement.createDataset.stepHeader"
        values={
          Object {
            "currentStepNumber": 0,
            "totalStepNumber": 0,
          }
        }
      />
    </h2>
  </EuiTitle>
  <EuiSpacer
    size="m"
  />
  <EuiFlexGroup>
    <EuiFlexItem>
      <EuiForm
        isInvalid={true}
      >
        <EuiCompressedFormRow
          describedByIds={Array []}
          display="rowCompressed"
          error={
            Array [
              "Input is invalid",
            ]
          }
          fullWidth={true}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          helpText={
            <React.Fragment>
              <FormattedMessage
                defaultMessage="Use an asterisk ({asterisk}) to match multiple indices."
                id="datasetManagement.createDataset.step.dataset.allowLabel"
                values={
                  Object {
                    "asterisk": <strong>
                      *
                    </strong>,
                  }
                }
              />
               
              <FormattedMessage
                defaultMessage="Spaces and the characters {characterList} are not allowed."
                id="datasetManagement.createDataset.step.dataset.disallowLabel"
                values={
                  Object {
                    "characterList": <strong>
                      %
                    </strong>,
                  }
                }
              />
            </React.Fragment>
          }
          isInvalid={true}
          label={
            <FormattedMessage
              defaultMessage="Index pattern name"
              id="datasetManagement.createDataset.step.datasetLabel"
              values={Object {}}
            />
          }
          labelType="label"
        >
          <EuiCompressedFieldText
            data-test-subj="createDatasetNameInput"
            fullWidth={true}
            isInvalid={true}
            name="dataset"
            onChange={[Function]}
            placeholder="index-name-*"
            value="%"
          />
        </EuiCompressedFormRow>
      </EuiForm>
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      <EuiCompressedFormRow
        describedByIds={Array []}
        display="rowCompressed"
        fullWidth={false}
        hasChildLabel={true}
        hasEmptyLabelSpace={true}
        labelType="label"
      >
        <EuiSmallButton
          data-test-subj="createDatasetGoToStep2Button"
          fill={true}
          iconSide="right"
          iconType="arrowRight"
          isDisabled={true}
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Next step"
            id="datasetManagement.createDataset.step.nextStepButton"
            values={Object {}}
          />
        </EuiSmallButton>
      </EuiCompressedFormRow>
    </EuiFlexItem>
  </EuiFlexGroup>
</div>
`;

exports[`Header should render normally 1`] = `
<div>
  <EuiTitle
    size="s"
  >
    <h2>
      <FormattedMessage
        defaultMessage="Step {currentStepNumber} of {totalStepNumber}: Define an index pattern"
        id="datasetManagement.createDataset.stepHeader"
        values={
          Object {
            "currentStepNumber": 0,
            "totalStepNumber": 0,
          }
        }
      />
    </h2>
  </EuiTitle>
  <EuiSpacer
    size="m"
  />
  <EuiFlexGroup>
    <EuiFlexItem>
      <EuiForm
        isInvalid={false}
      >
        <EuiCompressedFormRow
          describedByIds={Array []}
          display="rowCompressed"
          error={Array []}
          fullWidth={true}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          helpText={
            <React.Fragment>
              <FormattedMessage
                defaultMessage="Use an asterisk ({asterisk}) to match multiple indices."
                id="datasetManagement.createDataset.step.dataset.allowLabel"
                values={
                  Object {
                    "asterisk": <strong>
                      *
                    </strong>,
                  }
                }
              />
               
              <FormattedMessage
                defaultMessage="Spaces and the characters {characterList} are not allowed."
                id="datasetManagement.createDataset.step.dataset.disallowLabel"
                values={
                  Object {
                    "characterList": <strong>
                      %
                    </strong>,
                  }
                }
              />
            </React.Fragment>
          }
          isInvalid={false}
          label={
            <FormattedMessage
              defaultMessage="Index pattern name"
              id="datasetManagement.createDataset.step.datasetLabel"
              values={Object {}}
            />
          }
          labelType="label"
        >
          <EuiCompressedFieldText
            data-test-subj="createDatasetNameInput"
            fullWidth={true}
            isInvalid={false}
            name="dataset"
            onChange={[Function]}
            placeholder="index-name-*"
            value="k"
          />
        </EuiCompressedFormRow>
      </EuiForm>
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      <EuiCompressedFormRow
        describedByIds={Array []}
        display="rowCompressed"
        fullWidth={false}
        hasChildLabel={true}
        hasEmptyLabelSpace={true}
        labelType="label"
      >
        <EuiSmallButton
          data-test-subj="createDatasetGoToStep2Button"
          fill={true}
          iconSide="right"
          iconType="arrowRight"
          isDisabled={false}
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Next step"
            id="datasetManagement.createDataset.step.nextStepButton"
            values={Object {}}
          />
        </EuiSmallButton>
      </EuiCompressedFormRow>
    </EuiFlexItem>
  </EuiFlexGroup>
</div>
`;
